lib/std/math.zig

```diff
@@ -1284,6 +1284,7 @@ test "std.math.log2_int_ceil" {
 pub fn lossyCast(comptime T: type, value: anytype) T {
     switch (@typeInfo(T)) {
         .Float => {
+            if (isNan(value)) return nan(T);
```

Comment:
This case was already working as expected(`@floatCast(nan)` already returns `nan`) and there is no need to explicitely handle it.

Reply:
Didn't know that, I removed that line now.